actors/market/tests/harness.rs

```diff
@@ -0,0 +1,750 @@
+use fil_actors_runtime::{
+    test_utils::*, ActorError, SetMultimap, BURNT_FUNDS_ACTOR_ADDR, CRON_ACTOR_ADDR,
+    REWARD_ACTOR_ADDR, STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR, SYSTEM_ACTOR_ADDR,
+    VERIFIED_REGISTRY_ACTOR_ADDR,
+};
+use fvm_ipld_encoding::{to_vec, RawBytes};
+use fvm_shared::bigint::BigInt;
+use fvm_shared::clock::{ChainEpoch, EPOCH_UNDEFINED};
+use fvm_shared::crypto::signature::Signature;
+use fvm_shared::deal::DealID;
+use fvm_shared::piece::PaddedPieceSize;
+use fvm_shared::reward::ThisEpochRewardReturn;
+use fvm_shared::sector::StoragePower;
+use fvm_shared::smooth::FilterEstimate;
+use fvm_shared::{
+    address::Address, econ::TokenAmount, error::ExitCode, METHOD_CONSTRUCTOR, METHOD_SEND,
+};
+
+// Define common set of actor ids that will be used across all tests.
+pub const OWNER_ID: u64 = 101;
+pub const PROVIDER_ID: u64 = 102;
+pub const WORKER_ID: u64 = 103;
+pub const CLIENT_ID: u64 = 104;
+pub const CONTROL_ID: u64 = 200;
+
+pub const OWNER_ADDR: Address = Address::new_id(OWNER_ID);
+pub const PROVIDER_ADDR: Address = Address::new_id(PROVIDER_ID);
+pub const WORKER_ADDR: Address = Address::new_id(WORKER_ID);
+pub const CLIENT_ADDR: Address = Address::new_id(CLIENT_ID);
+pub const CONTROL_ADDR: Address = Address::new_id(CONTROL_ID);
+
+pub struct MinerAddresses {
+    pub owner: Address,
+    pub worker: Address,
+    pub provider: Address,
+    pub control: Vec<Address>,
+}
+
+// Use the predefined actor addresses by default
+impl Default for MinerAddresses {
+    fn default() -> Self {
+        MinerAddresses {
+            owner: OWNER_ADDR,
+            worker: WORKER_ADDR,
+            provider: PROVIDER_ADDR,
+            control: vec![CONTROL_ADDR],
+        }
+    }
+}
+
+pub fn setup() -> MockRuntime {
+    let mut actor_code_cids = HashMap::default();
+    actor_code_cids.insert(OWNER_ADDR, *ACCOUNT_ACTOR_CODE_ID);
+    actor_code_cids.insert(WORKER_ADDR, *ACCOUNT_ACTOR_CODE_ID);
+    actor_code_cids.insert(PROVIDER_ADDR, *MINER_ACTOR_CODE_ID);
+    actor_code_cids.insert(CLIENT_ADDR, *ACCOUNT_ACTOR_CODE_ID);
+
+    let mut rt = MockRuntime {
+        receiver: *STORAGE_MARKET_ACTOR_ADDR,
+        caller: *SYSTEM_ACTOR_ADDR,
+        caller_type: *INIT_ACTOR_CODE_ID,
+        actor_code_cids,
+        ..Default::default()
+    };
+    construct_and_verify(&mut rt);
+
+    rt
+}
+
+pub fn check_state(_rt: &mut MockRuntime) {
+    // TODO
+}
+
+pub fn construct_and_verify(rt: &mut MockRuntime) {
+    rt.expect_validate_caller_addr(vec![*SYSTEM_ACTOR_ADDR]);
+    assert_eq!(
+        RawBytes::default(),
+        rt.call::<MarketActor>(METHOD_CONSTRUCTOR, &RawBytes::default(),).unwrap()
+    );
+    rt.verify();
+}
+
+pub fn get_escrow_balance(rt: &MockRuntime, addr: &Address) -> Result<TokenAmount, ActorError> {
+    let st: State = rt.get_state();
+
+    let et = BalanceTable::from_root(rt.store(), &st.escrow_table).unwrap();
+
+    Ok(et.get(addr).unwrap())
```

Comment:
maybe an `expect` with helpful message would be nicer

Reply:
Fixed.